src/material.cpp

```diff
@@ -670,14 +670,136 @@ std::string pyne::Material::mcnp(std::string frac_type) {
       // Spaces are important for tests
       table_item = metadata["table_ids"][nucmcnp].asString();
       if (!table_item.empty()) {
-	oss << "     " << mcnp_id << "." << table_item << " ";
+  oss << "     " << mcnp_id << "." << table_item << " ";
       } else {
-	oss << "     " << mcnp_id << " ";
+  oss << "     " << mcnp_id << " ";
       }
       // The int needs a little formatting
       std::stringstream fs;
       fs << std::setprecision(4) << std::scientific << frac_sign << i->second \
-	 << std::endl;
+   << std::endl;
+      oss << fs.str();
+    }
+  }
+
+  return oss.str();
+}
+
+///---------------------------------------------------------------------------//
+std::string pyne::Material::phits(std::string frac_type) {
+  //////////////////// Begin card creation ///////////////////////
+  std::ostringstream oss;
+  // Material Card Header
+  oss << "[ M a t e r i a l ]" << std::endl;
+
+  // 'name'
+  if (metadata.isMember("name")) {
+    oss << "C name: " << metadata["name"].asString() << std::endl;
+  }
+  // Metadata comments
+  if (metadata.isMember("comments")) {
+    std::string comment_string = "comments: " + metadata["comments"].asString();
+    // Include as is if short enough
+    if (comment_string.length() <= 77) {
+      oss << "C " << comment_string << std::endl;
+    }
+    else { // otherwise create a remainder string and iterate/update it
+      oss << "C " << comment_string.substr(0,77) << std::endl;
+      std::string remainder_string = comment_string.substr(77);
+      while (remainder_string.length() > 77) {
+        oss << "C " << remainder_string.substr(0,77) << std::endl;
+        remainder_string.erase(0,77);
+      }
+      if (remainder_string.length() > 0) {
+        oss << "C " << remainder_string << std::endl;
+      }
+    }
+  }
+
+  // Metadata mat_num
+  oss << "M[ ";
+  if (metadata.isMember("mat_number")) {
+    int mat_num = metadata["mat_number"].asInt();
+    oss << mat_num;
+  } else {
+    oss << "?";
+  }
+  oss << " ]" << std::endl;
+
+  // check for metadata
+  if (metadata.isMember("GAS")){
+    oss << "     GAS=" << metadata["GAS"].asInt() << std::endl;
+  }
+  if (metadata.isMember("ESTEP")){
+    oss << "     ESTEP=" << metadata["ESTEP"].asInt() << std::endl;
+  }
+  if (metadata.isMember("NLIB")){
+    oss << "     NLIB=" << metadata["NLIB"].asInt() << std::endl;
+  }
+  if (metadata.isMember("PLIB")){
+    oss << "     PLIB=" << metadata["PLIB"].asInt() << std::endl;
+  }
+  if (metadata.isMember("ELIB")){
+    oss << "     ELIB=" << metadata["ELIB"].asInt() << std::endl;
+  }
+  if (metadata.isMember("HLIB")){
+    oss << "     HLIB=" << metadata["HLIB"].asInt() << std::endl;
+  }
+  // COND should be "<" or "=" or ">" if present
+  if (metadata.isMember("COND")){
+    oss << "     COND" << metadata["COND"].asString() << "0" << std::endl;
+  }
+
+  // Set up atom or mass frac map
+  std::map<int, double> fracs;
+  std::string frac_sign;
+
+  if ("atom" == frac_type) {
```

Comment:
If this is the same as MCNP, perhaps write a function that does this part and re-use